File: quicksort.py
# 1) Base: arrays com 0 e 1 elementos são considerados ordenados
# 2) Caso recursivo
# 3) Subarray de todos os elementos menores do que o pivô
# 4) Subarray de todos os elementos maiores do que o pivô


def quicksort(array):
    if len(array) < 2:
        return array
    else: 
        pivot = array[0]
        menores = [i for i in array[1:] if i <= pivot]
        maiores = [i for i in array[1:] if i > pivot]
        return quicksort(menores) + [pivot] + quicksort(maiores)
    
print(quicksort([10, 5, 2, 3]))

File: quicksort.js
function quicksort(array) {
    let menores = []
    let maiores = []
    if (array.lenght < 2) {
        return array;
    } else {
        pivot = array[0]
    for (i of array) {
        if (i <= pivot) {
            menores.push(i);
        } else {
            maiores.push(i);
        }
    }
        //let array2 = [...quicksort(menores), pivot, ...quicksort(maiores)];
        //return array2;
        console.log(menores, maiores, pivot)
    }
    // console.log(`Menor: ${menores} /// pivot: ${[pivot]} /// Maior:  ${maiores}`)
}

quicksort([10, 5, 2, 3])